File: python/add_digits_258.py
# TODO: Revisit this cause it would be nice to try and either just remember the digital root thing, or be able to do the naive solution
class Solution:
    def addDigits(self, num: int) -> int:
        if num == 0:
            return 0
        else:
            return 1 + (num-1) % 9
        

File: elixir/add_digits_258.ex
defmodule Solution do
  @spec add_digits(num :: integer) :: integer
  def add_digits(num) when num < 10, do: num

  def add_digits(num) do
    num
    |> Integer.digits()
    |> Enum.sum()
    |> add_digits()
  end
end

